ctapipe/image/modifications.py

```diff
@@ -0,0 +1,174 @@
+            continue
+
+        # remove light from current pixel
+        new_image[pixel] -= to_smear
+
+        # add light to neighbor pixels
+        neighbors = indices[indptr[pixel] : indptr[pixel + 1]]
+        n_neighbors = len(neighbors)
+
+        # all neighbors are equally likely to receive the charge
+        # we always distribute the charge into 6 neighbors, so that charge
+        # on the edges of the camera is lost
+        neighbor_charges = np.random.multinomial(to_smear, smear_probabilities)
+
+        for n in range(n_neighbors):
+            neighbor = neighbors[n]
+            new_image[neighbor] += neighbor_charges[n]
+
+    return new_image
+
+
+class ImageModifier(TelescopeComponent):
+    """
+    Abstract class for configurable image modifying algorithms. Use
+    ``ImageModifier.from_name()`` to construct an instance of a particular algorithm
+    """
+
+    @abstractmethod
+    def __call__(self, tel_id: int, image: np.ndarray) -> np.ndarray:
+        """
+        Modifies a given image
+
+        Returns
+        -------
+        np.ndarray
+            modified image
+        """
+
+
+class NSBNoiseAdder(ImageModifier):
+    """
+    Component to tune simulated background to
+    overserved NSB values.
+    A differentiation between bright and dim pixels is taking place
+    because this happens at DL1a level and in general the integration window
+    would change for peak-searching extraction algorithms with different background levels
+    introducing a bias to the charge in dim pixels.
+
+    The performed steps include:
+    - Smearing of the image (simulating a worse PSF)
+    - Adding poissonian noise (different for bright and dim pixels)
+    - Adding a bias to dim pixel charges (see above)
+    """
+
+    smear_factor = FloatTelescopeParameter(
+        default_value=0.0, help="Fraction of light to move to each neighbor"
+    ).tag(config=True)
+    transition_charge = FloatTelescopeParameter(
+        default_value=0.0, help="separation between dim and bright pixels"
+    ).tag(config=True)
+    dim_pixel_bias = FloatTelescopeParameter(
+        default_value=0.0, help="extra bias to add in dim pixels"
+    ).tag(config=True)
+    dim_pixel_noise = FloatTelescopeParameter(
+        default_value=0.0, help="expected extra noise in dim pixels"
+    ).tag(config=True)
+    bright_pixel_noise = FloatTelescopeParameter(
+        default_value=0.0, help="expected extra noise in bright pixels"
+    ).tag(config=True)
+    correct_bias = BoolTelescopeParameter(
+        default_value=True, help="If True subtract the expected noise from the image."
+    ).tag(config=True)
+
+    def __call__(self, tel_id, image, rng=None):
+        # TODO: how to get max_neighbors?
+        # This might be the safest way (including the decision if diagonal neighbors
+        # count for square pixels)
+        geom = self.subarray.tel[tel_id].camera.geometry
+        max_neighbors = geom.neighbor_matrix.sum(axis=1).max()
```

Comment:
This should be a property on the geometry and cached, not calculated for each pixel on each event.